functions/mqpublish.go

```diff
@@ -47,58 +47,79 @@ func Checkin(ctx context.Context, wg *sync.WaitGroup) {
+				if mqclient == nil || !mqclient.IsConnected() {
 					logger.Log(0, "MQ client is not connected, skipping checkin for server", server)
 					continue
 				}
 			}
-			for server, mqclient := range ServerSet {
-				if mqclient == nil {
-					logger.Log(0, "MQ client is not configured, skipping checkin for server", server)
-					continue
-				}
-			}
 			checkin()
 		}
 	}
 }
 
 func checkin() {
-
-	host := config.Netclient()
 	//should not be required
-	config.ReadNodeConfig()
-	config.ReadServerConf()
+	_ = config.ReadNodeConfig()
+	_ = config.ReadServerConf()
 	logger.Log(3, "checkin with server(s) for all networks")
-	for network, node := range config.GetNodes() {
+	var (
+		publicIP         string
+		err              error
+		shouldUpdateHost bool
+	)
+	for _, node := range config.GetNodes() {
+		node := node
 		server := config.GetServer(node.Server)
-		if node.Connected {
+		if node.Connected && len(publicIP) == 0 {
 			if !config.Netclient().IsStatic {
-				extIP, err := ncutils.GetPublicIP(server.API)
+				publicIP, err = ncutils.GetPublicIP(server.API)
 				if err != nil {
 					logger.Log(1, "error encountered checking public ip addresses: ", err.Error())
 				}
-				if config.Netclient().EndpointIP.String() != extIP && extIP != "" {
-					logger.Log(1, "network:", network, "endpoint has changed from ", config.Netclient().EndpointIP.String(), " to ", extIP)
-					config.Netclient().EndpointIP = net.ParseIP(extIP)
-					if err := PublishNodeUpdate(&node); err != nil {
-						logger.Log(0, "network:", network, "could not publish endpoint change")
-					}
+				if len(publicIP) > 0 && config.Netclient().EndpointIP.String() != publicIP {
+					logger.Log(1, "endpoint has changed from", config.Netclient().EndpointIP.String(), "to", publicIP)
+					config.Netclient().EndpointIP = net.ParseIP(publicIP)
+					shouldUpdateHost = true
 				}
 			}
 		}
-		//check version
-		//if node.Version != ncutils.Version {
-		//node.Version = ncutils.Version
-		//config.Write(&nodeCfg, nodeCfg.Network)
-		//}
-		Hello(&node)
 		if server.Is_EE && node.Connected {
-			logger.Log(0, "collecting metrics for node", host.Name)
+			logger.Log(0, "collecting metrics for network", node.Network)
 			publishMetrics(&node)
 		}
 	}
-	_ = UpdateHostSettings()
+	ip, err := getInterfaces()
+	if err != nil {
+		logger.Log(0, "failed to retrieve local interfaces during check-in", err.Error())
+	} else {
+		if ip != nil {
+			if len(*ip) != len(config.Netclient().Interfaces) {
+				config.Netclient().Interfaces = *ip
+			}
+		}
+	}
+	defaultInterface, err := getDefaultInterface()
+	if err != nil {
+		logger.Log(0, "default gateway not found", err.Error())
+	} else {
+		if defaultInterface != config.Netclient().DefaultInterface {
+			shouldUpdateHost = true
+			config.Netclient().DefaultInterface = defaultInterface
+		}
+	}
+
+	if shouldUpdateHost {
+		if err = UpdateHostSettings(); err != nil {
+			logger.Log(0, "failed to update host settings -", err.Error())
```

Comment:
`UpdateHostSettings` checks for port changes, this check here might avoid updating port changes